tqdm/autonotebook.py

```diff
@@ -21,7 +22,11 @@
 except Exception:
     from .std import tqdm, trange
 else:  # pragma: no cover
-    from .notebook import tqdm, trange
+    if "SPY_TESTING" in os.environ:
```

Comment:
- is `SPY_TESTING` always defined inside a spyder environment?
- is it fairly plausible for `SPY_TESTING` to be defined outside spyder?

Reply:
We have used this approach for many years without any issues. Checking environment variables of the form `SPY_XXXX` is also suggested in https://stackoverflow.com/questions/53829284/how-to-know-if-a-script-is-running-in-spyder-or-any-other-python-ide.

I will create an issue at the Spyder issue tracker to check this is a good approach.

Reply:
@casperdcl I have updated the detection method based on the suggestion by the Spyder devs. See https://github.com/spyder-ide/spyder/issues/21929